src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbol.cs

```diff
@@ -65,6 +65,7 @@ public MethodKind MethodKind
             public bool IsExtensionMethod
             {
                 get { return (_flags & IsExtensionMethodBit) != 0; }
+                set { _flags = value ? (_flags | IsExtensionMethodBit) : (_flags & ~IsExtensionMethodBit); }
```

Comment:
Upon further inspection, it looks like an immutability bug. Will file an issue to fix ReturnsVoid and undo this change.

---
In reply to: [123042381](https://github.com/dotnet/roslyn/pull/19703#discussion_r123042381) [](ancestors = 123042381,122884575)